### PR TITLE
improves IT suite 

### LINKED ISSUE #638: Improvements in IT Machine Controller Manager
**How to categorize this issue?**

/area testing
/kind enhancement
/priority 3

**What happened**:
Improvements in IT-suite required
**What you expected to happen**:
1. integration-test docs to run integration test locally should be modified to include new changes
2. v1 CRDs needs to be used 
3. MaxSurge of test-machineDeployments created should be increased to 2 as for Azure the test would timeout otherwise.
4. logic to know if string tags or key value pair tags are used , should be added.
5. machines created during tests should have different role tags as compared to usual machines created by MCM, so that orphan collection doesn't happen due to racing of MCM's in case when both control and target clusters are shoot clusters
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
**Environment**:

- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

### PR DESCRIPTION
**Pull request by himanshu-kun**
Improves IT framework

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #638 

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by prashanth26 — APPROVED** [review]
Thanks for this fix. 
/lgtm
cc: @rkjakeer - I hope this change looks good to you? And hoping earlier it was a mistake?

**Comment by prashanth26** [comment]
We need to refactor this now that the CRDs change is merged?

**Comment by himanshu-kun** [comment]
> We need to refactor this now that the CRDs change is merged?

Yes I have refactored it on my local branch , will push the changes by today.

File: pkg/test/integration/common/framework.go [hunk 1]
```diff
@@ -615,11 +692,46 @@ func (c *IntegrationTestFramework) SetupBeforeSuite() {
...
+	                - Scale down MCM running in seed to 0
+	                - Run MCM locally
+	    - If NO
+	        - Then, most likely this is a shoot whose original MCM is also running. Scale it down (this is challenging)
+	        - Apply the CRDs (Clone MCM repo before this as CRDs present in MCM repo)
+	        - Check if MC Image or MCM Image are set in the Makefile
+	            - If YES (that is, one of them or both of them are passed)
+	                - Prepare MCM Deployment in the control cluster by creating (/kubernetes/out-of-tree/deployment.yaml)
+	            - If NO (that is both of them are not passed)
+	                - Run MCM locally
```
Comment (AxiomSamarth) [inline pkg/test/integration/common/framework.go:715]: You can remove these reference comments once everything is working satisfactorily.

**Review by AxiomSamarth — CHANGES_REQUESTED** [review]
Looks good, @himanshu-kun I have done some nit picking. Also, if you remember then we had discussed to add a validation for tags in machineClass section [here](https://github.com/himanshu-kun/machine-controller-manager/blob/bb90d29a364fed00acee9a618fd70212318d2530/pkg/test/integration/common/helpers/handling_files.go#L120-L126) in handling_files.go so as to warn the user to use different tags so that the machines are not orphan collected. Can you introduce that check as well?

File: pkg/test/integration/common/framework.go [hunk 2]
```diff
@@ -1145,9 +1145,9 @@ func (c *IntegrationTestFramework) ControllerTests() {
 	})
 
 	// Testcase #03 | Orphaned Resources
-	ginkgo.Describe("orphaned resources excpet NICs", func() {
+	ginkgo.Describe("orphaned resources", func() {
 		ginkgo.Context("when the hyperscaler resources are querried", func() {
```
Comment (AxiomSamarth) [inline pkg/test/integration/common/framework.go:1149]:
```suggestion
		ginkgo.Context("when the hyperscaler resources are queried", func() {
```

File: pkg/test/integration/common/framework.go [hunk 3]
```diff
@@ -1145,9 +1145,9 @@ func (c *IntegrationTestFramework) ControllerTests() {
...
-			ginkgo.It("should have been deleted and orphan NICs be listed", func() {
+			ginkgo.It("should have been deleted", func() {
 				// if available, should delete orphaned resources in the cloud provider
 				ginkgo.By("Querrying and comparing")
```
Comment (AxiomSamarth) [inline pkg/test/integration/common/framework.go:1152]:
```suggestion
				ginkgo.By("Querying and comparing")
```

**Review by AxiomSamarth — APPROVED** [review]
/lgtm

**Review by prashanth26 — APPROVED** [review]
/lgtm post CI fix

**Review by AxiomSamarth — APPROVED** [review]
Thanks for the PR @himanshu-kun.

**Pull request merged**